lib/elastic_apm/instrumenter.rb

```diff
@@ -161,7 +166,8 @@ def start_span(
         name: name,
         subtype: subtype,
         action: action,
-        transaction_id: transaction.id,
+        transaction: transaction,
+        parent: parent,
         trace_context: trace_context || parent.trace_context.child,
```

Comment:
You could move the use of `parent.trace_context.child` to the `initialize` method of `Span`. I.e. just pass `trace_context` here and then in `Span#initialize` you can do: `@trace_context = trace_context || parent.trace_context.child`